Lesson27-Loops&Conditions/scripts/main.js

```diff
@@ -0,0 +1,301 @@
+    }
+// numberLength()
+
+
+// 5. Request 10 numbers from a user and count, how many are positive, negative, or zero.
+//      Also, count odd and even. Display the statistics. There’s only one variable (not 10) needed for input by a user. 
+// [[BUG: CODE RUNS AN EXTRA TIME, MAKING LENGTH OF ODD NUMBERS, 1 EXTRA]]
+function numberStats() {
+    // Empty variables to add arrays into        
+        let positiveNumbers = [];
+        let negativeNumbers = [];
+        let zeroNumbers = [];
+        let evenNumbers = [];
+        let oddNumbers = [];
+    // Asking for the numbers + Converting them to an Array
+        let userInput = prompt('Enter 10 numbers, sperated with a comma:');
+        let userInputSplit = userInput.split(',');
+        userInputArray = userInputSplit.map(Number);         
+    // For loop to check each number
+        for (i = 0; i <= userInputArray.length; i++) {   
+        //  Check for positive numbers
+            if (userInputArray[i] > 0) {
+                positiveNumbers.push(userInputArray[i])            
+            }    
+        //  Check for negative numbers
+            if (userInputArray[i] < 0) {
+                negativeNumbers.push(userInputArray[i])            
+            }     
+        //  Check for zero numbers
+            if (userInputArray[i] === 0) {
+                zeroNumbers.push(userInputArray[i])            
+            }  
+        // // Check for even numbers       
+            if (userInputArray[i] % 2 === 0) {
+                evenNumbers.push(userInputArray[i])            
+            }
+        // // Check for odd numbers       
+        if (userInputArray[i] % 2 != 0) {
+            oddNumbers.push(userInputArray[i])            
+        }    
+    }
+    // Printing statements 
+        console.log('Your entered list was ' + userInputArray);
+        console.log('All the positive numbers in your list is: ' + positiveNumbers);
+        console.log('The length of your positive number list is: ' + positiveNumbers.length);
+        console.log('All the negative numbers in your list is: ' + negativeNumbers);
+        console.log('The length of your negative number list is: ' + negativeNumbers.length);
+        console.log('All the zero numbers in your list is: ' + zeroNumbers);
+        console.log('The length of your zero number list is: ' + zeroNumbers.length);
+        console.log('All the even numbers in your list is: ' + evenNumbers);
+        console.log('The length of your even number list is: ' + evenNumbers.length);
+        console.log('All the odd numbers in your list is: ' + oddNumbers);
+        console.log('The length of your odd number list is: ' + oddNumbers.length);
+    }
+    // numberStats()
+
+
+// 6. Loop a calculator. Request 2 numbers and a sign, solve the problem, display the result and ask if the user wants one more.
+//      The loop continues until the user refuses. 
+//[[BUGG: MULTIPLY DOESN'T MULTIPLY]]
```

Comment:
What is a bug?

Reply:
The code doesn't work properly. 
It works until you choose the multiplication option, then it returns a different value. It doesn't multiply the values.